qiskit/algorithms/minimum_eigen_solvers/adaptvqe.py

```diff
@@ -0,0 +1,332 @@
+            )
+            # compute gradient
+            state_grad = self.gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            value_dict = {param_sets[-1]: 0.0}
+            state_grad_result = sampler.convert(state_grad, params=value_dict).eval()
+            logger.info("Gradient computed : %s", str(state_grad_result))
+            res.append((np.abs(state_grad_result[-1]), exc))
+        return res, expectation
+
+    @staticmethod
+    def _check_cyclicity(indices: List[int]) -> bool:
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: The list of chosen gradient indices.
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    def solve(
+        self,
+        aux_operators: Optional[ListOrDict[PauliSumOp]] = None,
+    ) -> AdaptVQEResult:
+        """Computes the ground state.
+
+        Args:
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: if a solver other than VQE or a ansatz other than UCCSD is provided
+                or if the algorithm finishes due to an unforeseen reason.
+            ValueError: If the grouped property object returned by the driver does not contain a
+                main property as requested by the problem being solved (`problem.main_property_name`)
+            QiskitError: If the user-provided `aux_operators` contain a name which clashes
+                with an internally constructed auxiliary operator. Note: the names used for the
+                internal auxiliary operators correspond to the `Property.name` attributes which
+                generated the respective operators.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An AdaptVQEResult which is an ElectronicStructureResult but also includes runtime
+            information about the AdaptVQE algorithm like the number of iterations, finishing
+            criterion, and the final maximum gradient.
+        """
+        # if not isinstance(self.ansatz, EvolvedOperatorAnsatz):
+        #     raise QiskitNatureError("The AdaptVQE algorithm requires the use of the evolved operator ansatz")
+
+        # We construct the ansatz once to be able to extract the full set of excitation operators.
+        self._tmp_ansatz._build()
+        self._excitation_pool = copy.deepcopy(self._tmp_ansatz.operators)
+
+        threshold_satisfied = False
+        alternating_sequence = False
+        max_iterations_exceeded = False
```

Comment:
These three variables are effectively a boolean encoding of the information stored in `finishing_criterion`. However, besides being set to `True` once, these are not being used. Thus, I suggest that we remove them entirely